File: android-app/app/src/main/java/com/borehole/management/data/local/entity/SurveyEntity.kt
package com.borehole.management.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "surveys")
data class SurveyEntity(
    @PrimaryKey
    val localId: String, // UUID for local storage
    val serverId: String? = null, // Server ID after sync
    val jobId: String,
    val surveyorId: String,
    val gpsLat: Double,
    val gpsLng: Double,
    val gpsAccuracy: Float? = null,
    val gpsCapturedAt: Date,
    val recommendedMinDepth: Float,
    val recommendedMaxDepth: Float,
    val expectedBreaks: String, // JSON array as string
    val soilType: String? = null,
    val groundConditions: String? = null,
    val surveyMethod: String? = null,
    val graphFileLocalPath: String? = null,
    val graphFileServerId: String? = null,
    val disclaimerAck: Boolean,
    val surveyorNotes: String? = null,
    val deviceId: String,
    val offlineCreatedAt: Date,
    val synced: Boolean = false,
    val syncStatus: SyncStatus = SyncStatus.PENDING,
    val syncAttempts: Int = 0,
    val lastSyncError: String? = null,
    val createdAt: Date,
    val updatedAt: Date
)

enum class SyncStatus {
    PENDING,
    IN_PROGRESS,
    SYNCED,
    FAILED,
    CONFLICT
}

File: backend/src/surveys/survey.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from 'typeorm';
import { User } from '../users/user.entity';
import { Job } from '../jobs/job.entity';

@Entity('survey_reports')
export class SurveyReport {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => Job)
  @JoinColumn({ name: 'jobId' })
  job: Job;

  @Column()
  jobId: string;

  @ManyToOne(() => User)
  @JoinColumn({ name: 'surveyorId' })
  surveyor: User;

  @Column()
  surveyorId: string;

  // GPS Data
  @Column('decimal', { precision: 10, scale: 8 })
  gpsLat: number;

  @Column('decimal', { precision: 11, scale: 8 })
  gpsLng: number;

  @Column('decimal', { precision: 5, scale: 2, nullable: true })
  gpsAccuracy: number;

  @Column({ nullable: true })
  gpsCapturedAt: Date;

  // Survey Data
  @Column('decimal', { precision: 6, scale: 2 })
  recommendedMinDepth: number;

  @Column('decimal', { precision: 6, scale: 2 })
  recommendedMaxDepth: number;

  @Column('simple-array')
  expectedBreaks: number[];

  @Column({ nullable: true })
  soilType: string;

  @Column({ type: 'text', nullable: true })
  groundConditions: string;

  @Column({ nullable: true })
  surveyMethod: string;

  // Graph/resistivity file
  @Column({ nullable: true })
  graphFileId: string;

  // Quality assurance
  @Column({ default: false })
  disclaimerAck: boolean;

  @Column({ type: 'text', nullable: true })
  surveyorNotes: string;

  // Offline sync fields
  @Column({ nullable: true })
  deviceId: string;

  @Column({ nullable: true })
  offlineCreatedAt: Date;

  @Column({ default: false })
  synced: boolean;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}